src/Symfony/Component/Yaml/Parser.php

```diff
@@ -386,6 +386,9 @@ private function doParse(string $value, int $flags)
                     $value = '';
 
                     foreach ($this->lines as $line) {
+                        if ('' !== $line && '#' === $line[0]) {
```

Comment:
Do we need to trim first? If I don't miss anything the following YAML string should be parsed the same:

```yaml
parameters:
    abc
  # Comment
```

Reply:
I added trim :+1: